### PR TITLE
Chain ID and FeeHistory Tweaks

### PR DESCRIPTION
**Pull request by shemnon**
* Support reward percentages for FeeHistory
* re-jigger chain_id detection, drive from network if not hard wired
* prettier reformatting

### CODE REVIEW

File: packages/relay/src/index.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export interface Eth {
   getUncleCountByBlockNumber(): Promise<string>;
 
   // getWork();
-  feeHistory(): Promise<any>;
+  feeHistory(blockCount: string, newestBlock: string, rewardPercentiles: Array<number>|null): Promise<any>;
```
Comment (Nana-EC) [inline packages/relay/src/index.ts:89]:
```suggestion
  feeHistory(blockCount: number, newestBlock: string, rewardPercentiles: Array<number>|null): Promise<any>;
```

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -64,42 +64,47 @@ export class EthImpl implements Eth {
...
     this.mirrorNode = mirrorNode;
     this.mirrorNodeClient = mirrorNodeClient;
     this.logger = logger;
 
     // Compute the chainId
     // FIXME: Should default to a "dev net" number. Is this it?
-    const configuredChainId = process.env.CHAIN_ID || '298';
-    this.chain = EthImpl.prepend0x(Number(configuredChainId).toString(16));
-    this.logger.info("Running with chainId=%s", this.chain);
+    //const configuredChainId = process.env.CHAIN_ID || '298';
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:105]:
```suggestion
```

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -114,21 +119,30 @@ export class EthImpl implements Eth {
   /**
    * Gets the fee history.
    */
-  async feeHistory() {
+  async feeHistory(
+    blockCount: string,
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:123]:
```suggestion
    blockCount: number,
```

File: packages/relay/src/lib/eth.ts [hunk 4]
```diff
@@ -417,32 +479,55 @@ export class EthImpl implements Eth {
...
       // Get a reasonable value for "gas" if it is not specified.
-      const gas: number = call.gas == null
-          ? 400_000
-          : typeof call.gas === 'string' ? Number(call.gas) : call.gas;
+      let gas: number;
+      if (typeof call.gas === 'string') {
+        gas = Number(call.gas);
+      } else {
+        if (call.gas == null) {
+          gas = 400_000;
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:497]: We should make this configureable i.e. add to the env file.
So should a change be needed in production for some reason it doesn't require re-deployments

File: packages/relay/src/lib/mirrorNode.ts [hunk 5]
```diff
@@ -74,11 +74,20 @@ export class MirrorNode {
     }
     this.logger.debug('Computing fee history based on the last %d blocks', mostRecentBlocks.length);
 
-    return {
-      baseFeePerGas: Array(mostRecentBlocks.length).fill('0x' + fee.toString(16)),
-      gasUsedRatio: Array(mostRecentBlocks.length).fill('0.5'),
-      oldestBlock: mostRecentBlocks[0].number
-    };
+    if (rewardPercentiles) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/mirrorNode.ts:77]: Can simplify to 
```
  const feeHistoryResponse = {
    baseFeePerGasArray: Array(mostRecentBlocks.length).fill('0x' + fee.toString(16)),
    gasUsedRatioArray: Array(mostRecentBlocks.length).fill('0.5'),
    oldestBlockNumber: mostRecentBlocks[0].number
  };

  if (rewardPercentiles) {
    feeHistoryResponse['reward'] = Array(mostRecentBlocks.length).fill(Array(rewardPercentiles.length).fill("0x0"));
  } 

  return feeHistoryResponse;
```

File: packages/relay/src/lib/relay.ts [hunk 6]
```diff
@@ -65,22 +81,27 @@ export class RelayImpl implements Relay {
     return this.ethImpl;
   }
 
-  initClient(type: string | null = null): Client {
-    dotenv.config({ path: findConfig('.env') || '' });
-    const hederaNetwork: string = process.env.HEDERA_NETWORK || '{}';
+  initClient(hederaNetwork: string, type: string | null = null): Client {
     let client: Client;
-    if (hederaNetwork in ['mainnet', 'testnet', 'previewnet']) {
+    if (hederaNetwork in RelayImpl.chainIds) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/relay.ts:86]:
```suggestion
    if (hederaNetwork.toLowerCase() in RelayImpl.chainIds) {
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Some minor improvement suggestions

File: packages/relay/src/lib/eth.ts (hunk 4 above)
Comment (shemnon) [reply packages/relay/src/lib/eth.ts:497]: in another pr.  I was dodging sonar complaints here.

File: packages/relay/src/index.ts (hunk 1 above)
Comment (shemnon) [reply packages/relay/src/index.ts]: done

File: packages/relay/src/lib/eth.ts (hunk 2 above)
Comment (shemnon) [reply packages/relay/src/lib/eth.ts]: Done.  Moved log line to where we calculate the chain id.

File: packages/relay/src/lib/eth.ts (hunk 3 above)
Comment (shemnon) [reply packages/relay/src/lib/eth.ts]: done

File: packages/relay/src/lib/relay.ts (hunk 6 above)
Comment (shemnon) [reply packages/relay/src/lib/relay.ts]: done

File: packages/relay/src/lib/mirrorNode.ts (hunk 5 above)
Comment (shemnon) [reply packages/relay/src/lib/mirrorNode.ts:83]: done

File: packages/relay/src/lib/eth.ts (hunk 4 above)
Comment (Nana-EC) [reply packages/relay/src/lib/eth.ts:499]: Sounds good

**Pull request merged**